src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1468,6 +1466,33 @@ def resolve_external_symbols(
         resolved_imports = {**imported}
         resolved_values_list = list(nonlocals.items())
 
+        # Resolve function-like imports recursively
+        func_externals = dict(
+            filter(
+                lambda name_value: isinstance(name_value[1], types.FunctionType), context.items()
+            )
+        )
+        while func_externals:
+            new_func_externals = {}
+            for name, value in func_externals.items():
+                # Annotate if not already done
+                if isinstance(value, types.FunctionType) and not hasattr(value, "_gtscript_"):
+                    GTScriptParser.annotate_definition(value)
+
+                # Get sub-context
+                sub_context, sub_unbound = gt_meta.get_closure(
+                    value, included_nonlocals=True, include_builtins=False
+                )
+
+                # Resolve imports or add to list to recursively resolve
+                for imported_name, imported_value in value._gtscript_["imported"].items():
+                    if isinstance(imported_value, types.FunctionType):
+                        add_func_externals[imported_name] = sub_context[imported_value]
```

Comment:
I guess this should be:

```suggestion
                        new_func_externals[imported_name] = sub_context[imported_value]
```

If the tests are passing even with this typo in the code, there is something else wrong either in the code or in the tests.